src/main/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAligner.scala

```diff
@@ -116,66 +136,132 @@ class NeedlemanWunschAligner(val scoringFunction: (Byte,Byte) => Int,
     *
     * @param query the query sequence
     * @param target the target sequence
-    * @return a tuple of the scoring matrix and the traceback matrix
+    * @return an array of alignment matrices, where the indices to the array are the Directions
     */
-  protected def buildMatrices(query: Array[Byte], target: Array[Byte]): (Matrix[Int], Matrix[Direction]) = {
-    val scoring = Matrix[Int](query.length + 1, target.length + 1)
-    val trace   = Matrix[Direction](query.length + 1, target.length + 1)
+  protected def buildMatrices(query: Array[Byte], target: Array[Byte]): Array[AlignmentMatrix] = {
+    val matrices = Range.inclusive(0, 2).map(i => AlignmentMatrix(direction=i.toByte, queryLength=query.length, targetLength=target.length)).toArray
```

Comment:
Why `Range.inclusive` and not `AllDirections`?

Reply:
Doh.  Because I did something else first and didn't revisit.  Fixed.